web/src/app/administration/components/workbasket-overview/workbasket-overview.component.spec.ts

```diff
@@ -0,0 +1,176 @@
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { DomainService } from '../../../shared/services/domain/domain.service';
+import { RouterTestingModule } from '@angular/router/testing';
+import { RequestInProgressService } from '../../../shared/services/request-in-progress/request-in-progress.service';
+import { SelectedRouteService } from '../../../shared/services/selected-route/selected-route';
+import { NotificationService } from '../../../shared/services/notifications/notification.service';
+import { ActivatedRoute } from '@angular/router';
+import { CreateWorkbasket, SelectWorkbasket } from '../../../shared/store/workbasket-store/workbasket.actions';
+
+const showDialogFn = jest.fn().mockReturnValue(true);
+const NotificationServiceSpy = jest.fn().mockImplementation(
+  (): Partial<NotificationService> => ({
+    triggerError: showDialogFn,
+    showToast: showDialogFn
+  })
+);
+
+const mockActivatedRoute = {
+  firstChild: {
+    params: of({
+      id: 'new-workbasket'
+    })
+  }
+};
+
+const mockActivatedRouteAlternative = {
+  firstChild: {
+    params: of({
+      id: '101'
+    })
+  }
+};
+
+const mockActivatedRouteNoParams = {};
+
+@Component({ selector: 'taskana-administration-workbasket-list', template: '' })
+class WorkbasketListStub {}
+
+@Component({ selector: 'taskana-administration-workbasket-details', template: '' })
+class WorkbasketDetailsStub {}
+
+@Component({ selector: 'svg-icon', template: '' })
+class SvgIconStub {}
+
+describe('WorkbasketOverviewComponent', () => {
+  let fixture: ComponentFixture<WorkbasketOverviewComponent>;
+  let debugElement: DebugElement;
+  let component: WorkbasketOverviewComponent;
+  let store: Store;
+  let actions$: Observable<any>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule, RouterTestingModule.withRoutes([]), NgxsModule.forRoot([WorkbasketState])],
+      declarations: [WorkbasketOverviewComponent, WorkbasketListStub, WorkbasketDetailsStub, SvgIconStub],
+      providers: [
+        WorkbasketService,
+        { provide: NotificationService, useClass: NotificationServiceSpy },
+        { provide: ActivatedRoute, useValue: mockActivatedRoute },
+        DomainService,
+        RequestInProgressService,
+        SelectedRouteService
+      ]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(WorkbasketOverviewComponent);
+    debugElement = fixture.debugElement;
+    component = fixture.debugElement.componentInstance;
+    store = TestBed.inject(Store);
+    actions$ = TestBed.inject(Actions);
+    fixture.detectChanges();
+  }));
+
+  it('should create the component', () => {
+    let activatedRoute = TestBed.inject(ActivatedRoute);
+    expect(component).toBeTruthy();
+    activatedRoute.firstChild.params.subscribe((params) => {
+      expect(component.routerParams).toMatchObject(params);
```

Comment:
Missing async / done

Reply:
removed this expect, this was for debugging purpose